### PR TITLE
Added split test into template plugin

### PR DESCRIPTION
**Pull request by apankratovantonp**
Split test should pass for all configuration parameter

### CODE REVIEW

File: docs/template_plugin/src/CMakeLists.txt [hunk 1]
```diff
@@ -37,8 +37,8 @@ target_link_libraries(${TARGET_NAME} PRIVATE
     interpreter_backend)
 
 # ATTENTION: uncomment to register a plugin in the plugins.xml file
-# ie_register_plugins(MAIN_TARGET ${TARGET_NAME}
-#                     POSSIBLE_PLUGINS ${TARGET_NAME})
+ie_register_plugins(MAIN_TARGET ${TARGET_NAME}
+                    POSSIBLE_PLUGINS ${TARGET_NAME})
```
Comment (ilya-lavrenov) [inline docs/template_plugin/src/CMakeLists.txt:41]: needed?

File: docs/template_plugin/src/template_executable_network.cpp [hunk 2]
```diff
@@ -62,6 +62,7 @@ void TemplatePlugin::ExecutableNetwork::CompileNetwork(const std::shared_ptr<con
         if (previousOutput.get_node()->get_output_size() > 1) {
             outputName += '.' + std::to_string(previousOutput.get_index());
         }
+        std::cout << result->get_friendly_name() << " " << outputName << " " << previousOutput.get_index() << std::endl;
```
Comment (ilya-lavrenov) [inline docs/template_plugin/src/template_executable_network.cpp:65]: needed?

File: docs/template_plugin/src/CMakeLists.txt (hunk 1 above)
Comment (apankratovantonp) [reply docs/template_plugin/src/CMakeLists.txt:41]: Will be removed. This PR should reproduce some problems in ngraph. When the problem are solved all this things will be removed

File: docs/template_plugin/src/template_executable_network.cpp (hunk 2 above)
Comment (apankratovantonp) [reply docs/template_plugin/src/template_executable_network.cpp:65]: same as above

**Comment by vurusovs** [comment]
please, rebase to the latest master to fix stress tests failure

File: docs/template_plugin/tests/functional/shared_tests_instances/single_layer_tests/split.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32
+};
+
+INSTANTIATE_TEST_CASE_P(NumSplitsCheck, SplitLayerTest,
+                        ::testing::Combine(
+                                ::testing::Values(1, 2, 3, 5, 6, 10, 30),
+                                ::testing::Values(0, 1, 2, 3),
+                                ::testing::ValuesIn(netPrecisions),
```
Comment (iefode) [inline docs/template_plugin/tests/functional/shared_tests_instances/single_layer_tests/split.cpp:22]: Minor comment, could you use one style in parameters? I am talking about the vector `netPrecisions`

Comment (apankratovantonp) [reply docs/template_plugin/tests/functional/shared_tests_instances/single_layer_tests/split.cpp:22]: What should be changed. Could you explain, please?

Comment (iefode) [reply docs/template_plugin/tests/functional/shared_tests_instances/single_layer_tests/split.cpp:22]: I mean `::testing::ValuesIn(InferenceEngine::Precision::FP32),`

Comment (apankratovantonp) [reply docs/template_plugin/tests/functional/shared_tests_instances/single_layer_tests/split.cpp]: Done

File: docs/template_plugin/src/template_executable_network.cpp (hunk 2 above)
Comment (apankratovantonp) [reply docs/template_plugin/src/template_executable_network.cpp]: removed

**Pull request merged**